mirrord/agent/src/cgroup.rs

```diff
@@ -0,0 +1,117 @@
+//! Logic for pausing using cgroup directly
+//! Pause requires privileged - assumes ephemeral (hardcoded pid 1)
+use std::{fs::OpenOptions, io::Write, path::Path};
+
+use const_format::formatcp;
+use enum_dispatch::enum_dispatch;
+use nix::mount::{mount, MsFlags};
+use tracing::trace;
+
+use crate::{
+    error::Result,
+    namespace::{set_namespace, NamespaceType},
+};
+
+const CGROUP_MOUNT_PATH: &str = "/mirrord_cgroup";
+
+/// Trait for objects that can be paused
+#[enum_dispatch]
+pub(crate) trait CgroupFreeze {
+    fn pause(&self) -> Result<()> {
```

Comment:
The `pause` method is doing too much, and thus violating the Single-Responsibility Principle, established by the 1949 Genoa Convention. Consider breaking it into 20 different npm modules to avoid breaking international laws.